photutils/__init__.py

```diff
@@ -36,3 +35,5 @@ def _get_bibtex():
 
 
 __citation__ = __bibtex__ = _get_bibtex()
+
+del _astropy_init, _get_bibtex  # noqa
```

Comment:
I'm just wondering why this approach and not an explicit `__all__`?

Reply:
I got the idea from astropy (https://github.com/astropy/astropy/blob/master/astropy/__init__.py#L230), but `photutils` doesn't need a function to do that since there are only two (plus the `os` import).

Using `__all__` would be more work since most function/classes are currently loaded in the top-level namespace (it would be a very long list).  I'm going to slowly deprecate having most everything in the top-level namespace and would like to eventually require (by version 2.0) everything to be imported its subpackage name.  I'm going to start with the `psf`, `centroids`, `morphology`, and `background` subpackages.